tests/sample_size/test_sample_size_calculator.py

```diff
@@ -0,0 +1,218 @@
+
+    def test_register_ratio_metric(self):
+        test_numerator_mean = 2000
+        test_numerator_variance = 100000
+        test_denominator_mean = 200
+        test_denominator_variance = 2000
+        test_covariance = 5000
+        test_mde = 10
+        test_variance = 5
+
+        calculator = SampleSizeCalculator()
+        calculator.register_ratio_metric(
+            test_numerator_mean,
+            test_numerator_variance,
+            test_denominator_mean,
+            test_denominator_variance,
+            test_covariance,
+            test_mde,
+        )
+
+        self.assertEqual(len(calculator.ratio_metrics), 1)
+        self.assertEqual(calculator.ratio_metrics[0].variance, test_variance)
+        self.assertEqual(calculator.ratio_metrics[0].mde, test_mde)
+        self.assertEqual(calculator.boolean_metrics, [])
+        self.assertEqual(calculator.numeric_metrics, [])
+
+        calculator.register_ratio_metric(
+            test_numerator_mean,
+            test_numerator_variance,
+            test_denominator_mean,
+            test_denominator_variance,
+            test_covariance,
+            test_mde,
+        )
+        self.assertEqual(len(calculator.ratio_metrics), 2)
+
+    @patch("statsmodels.stats.power.NormalIndPower.solve_power")
+    def test_get_single_sample_size_normal(self, mock_solve_power):
+        test_probability = 0.05
+        test_mde = 0.02
+        test_sample_size = 2000
+        test_metric = BooleanMetric(
+            test_probability,
+            test_mde,
+        )
+        mock_solve_power.return_value = test_sample_size
+
+        calculator = SampleSizeCalculator()
+        sample_size = calculator._get_single_sample_size(test_metric)
+
+        self.assertEqual(sample_size, test_sample_size)
+        mock_solve_power.assert_called_once_with(
+            effect_size=0.09176629354822471,
+            alpha=DEFAULT_ALPHA,
+            power=DEFAULT_POWER,
+            ratio=1,
+            alternative="two-sided",
+        )
+
+    @patch("statsmodels.stats.power.TTestIndPower.solve_power")
+    def test_get_single_sample_size_ttest(self, mock_solve_power):
+        test_variance = 1000
+        test_mde = 5
+        test_sample_size = 2000
+        test_metric = NumericMetric(
+            test_variance,
+            test_mde,
+        )
+        mock_solve_power.return_value = test_sample_size
+        calculator = SampleSizeCalculator()
+
+        sample_size = calculator._get_single_sample_size(test_metric)
+
+        self.assertEqual(sample_size, test_sample_size)
+        mock_solve_power.assert_called_once_with(
+            effect_size=0.15811388300841897,
+            alpha=DEFAULT_ALPHA,
+            power=DEFAULT_POWER,
+            ratio=1,
+            alternative="two-sided",
+        )
+
+    @patch("sample_size.sample_size_calculator.SampleSizeCalculator._get_single_sample_size")
+    def test_get_overall_sample_size_bool(self, mock_get_single_sample_size):
```

Comment:
nit:[ add comment reminding future contributors to parameterize these](https://github.com/godaddy/sample-size/pull/4#discussion_r761609769)